### PR TITLE
Added support for NegatedPropertySet property path

### PR DESCRIPTION
**Pull request by umbreak**

### CODE REVIEW

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala [hunk 1]
```diff
@@ -82,6 +90,10 @@ object PropPath extends PropPathBuilder {
     case PathZeroOrOne(uri)                => s"(${uri.show})?"
     case SeqPath(first, second)            => s"${showPath.show(first)}/${showPath.show(second)}"
     case AlternativeSeqPath(first, second) => s"${showPath.show(first)}|${showPath.show(second)}"
+    case NegatedSeqPath(Nil)               => ""
+    case NegatedSeqPath(list) =>
+      list.foldLeft(Vector.empty[String])((acc, current) => acc :+ showPath.show(current)).mkString("!(", "|", ")")
```
Comment (bogdanromanx) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala]: format folds with curlies, easier to read:

```
  val paths = list.foldLeft(Vector.empty[String]) {
    case (acc, el) => acc :+ current.show
  }
  paths.mkstring("!(", "|", ")")
```

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala [hunk 2]
```diff
@@ -111,28 +123,49 @@ trait PropPathBuilder {
       }
     }
 
+    def many(paths: List[Path]): Either[PropPathError, PropPath] = {
+      Try {
```
Comment (bogdanromanx) [inline modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala]: No need to `Try` and throw, you can directly foldleft over eithers and compose them with flatMap (`import cats.syntax.either._`)

Comment (umbreak) [reply modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala]: But I want to stop the fold loop whenever I see the first error. One error in any of the paths will mean the whole thing fails. And throwing scapes from the fold loop

File: modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala (hunk 1 above)
Comment (umbreak) [reply modules/indexing/src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/filtering/PropPath.scala]: Done

**Comment by umbreak** [comment]
retest this please

**Comment by umbreak** [comment]
retest this please

**Comment by umbreak** [comment]
retest this please

**Pull request merged**